packages/api/lib/FileUtils.js

```diff
@@ -99,15 +106,15 @@ const filterDatabaseProperties = (file) =>
     Object.keys(schemas.granule.properties.files.items.properties)
   );
 
-const buildDatabaseFile = (providerURL, file) =>
+const buildDatabaseFile = (s3, providerURL, file) =>
   flow([
     setBucket,
     setKey,
     setChecksum,
     setFileName,
     partial(setSource, providerURL),
     parseSource,
-    setS3FileSize // This one is last because it returns a Promise
+    partial(setS3FileSize, s3) // This one is last because it returns a Promise
```

Comment:
While I do like to see your use of some "functional" functions, I'm not crazy about the use of `flow` with all of the setter functions that each create a new object, but I'm not going to hold up approval because of it.

Just for future reference, take a look at Ramda's `applySpec` function. Here's how I might construct things with it:

```javascript
import { applySpec } from 'ramda';

setS3FileSize(s3, parseSource(applySpec({
  bucket: getBucket,
  key: getKey,
  checksum: getChecksum,
  fileName: getFileName,
  source: partial(buildFileSourceURL, providerURL)
})(file)));
```